### PR TITLE
Fix nokigiri with new version of ruby

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

**Comment by TApplencourt** [comment]
Should fix servesh issue (who is magicaly able to install new version of Ruby. I  guess because somebody else install rust)

File: packages/ruby-nokogiri/package.py [hunk 1]
```diff
@@ -14,8 +14,15 @@ class RubyNokogiri(RubyPackage):
...
     url      = "https://rubygems.org/downloads/nokogiri-1.12.5.gem"
 
+    version('1.16.7', sha256='f819cbfdfb0a7b19c9c52c6f2ca63df0e58a6125f4f139707b586b9511d7fe95', expand=False)
     version('1.12.5', sha256='2b20905942acc580697c8c496d0d1672ab617facb9d30d156b3c7676e67902ec', expand=False)
 
-    depends_on('ruby@2.5.0:', type=('build', 'run'))
-    depends_on('ruby-racc@1.8.1:', type=('build','run'))
-    depends_on('ruby-mini-portile2@2.6.1', type=('build', 'run'))
+    with when('@1.12.5'):
+       depends_on('ruby@2.5.0:3.0.0', type=('build', 'run'))
```
Comment (Kerilk) [inline packages/ruby-nokogiri/package.py:21]:
```suggestion
       depends_on('ruby@2.5.0:2.7.999', type=('build', 'run'))
```
or is it working with 3.0.0?

File: packages/ruby-nokogiri/package.py [hunk 2]
```diff
@@ -14,8 +14,15 @@ class RubyNokogiri(RubyPackage):
...
+       depends_on('ruby-racc@1.4.0:', type=('build','run'))
+       depends_on('ruby-mini-portile2@2.6.1:2.7.0', type=('build', 'run'))
+
+    with when('@1.16.7'):
+       depends_on('ruby@3.0.0:', type=('build', 'run'))
+       depends_on('ruby-racc@1.4.0:', type=('build','run'))
```
Comment (Kerilk) [inline packages/ruby-nokogiri/package.py:27]: This one seems to be the same for both

File: packages/ruby-nokogiri/package.py (hunk 1 above)
Comment (TApplencourt) [reply packages/ruby-nokogiri/package.py:21]: I don't know.  Just know that:
- 1.17 need ruby3+,
- 1.12 need ruby 2.5+
- servesh saw 1.12 failed with ruby 3.3.
- CI passed before with 1.12 and 3.1.

I assume we can do `ruby 3.1`, but then we have no way of testing it. 
So maybe it's cleaner to indeed do `2.7.99`, aka all versions before Ruby 3. It's a little to restrictive but 🤷🏽

File: packages/ruby-nokogiri/package.py (hunk 2 above)
Comment (TApplencourt) [reply packages/ruby-nokogiri/package.py:27]: Yeah, I just copied and pasted the `https://rubygems.org/gems/nokogiri/versions/1.16.7/dependencies`, but we can refactor it.

Comment (TApplencourt) [reply packages/ruby-nokogiri/package.py:27]: Sorry here: https://rubygems.org/gems/nokogiri/

**Review by Kerilk — DISMISSED** [review]

**Review by Kerilk — APPROVED** [review]

**Pull request merged**